File: python/rna-transcription/rna_transcription.py
translations = {
    'G' : 'C',
    'C' : 'G',
    'T' : 'A',
    'A' : 'U',
}

def to_rna(dna_strand):
    return "".join(map(lambda x: translations[x] , dna_strand))

File: typescript/rna-transcription/rna-transcription.ts
class Transcriptor {
    public toRna( dna : string ): string {
        let rna = "";
        [...dna].forEach(letter => {
            if (this.translations[letter] === undefined) {
                throw new Error("Invalid input DNA.");
            }
            rna += this.translations[letter];
        })
        return rna;
    }

    private translations: { [key: string]: string } = {
        "G": "C",
        "C": "G",
        "T": "A",
        "A": "U"
      }
}

export default Transcriptor